x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_columns/use_columns.ts

```diff
@@ -182,21 +182,40 @@ export const useColumns = ({
 
   const defaultColumnsRef = useRef<typeof defaultColumns>(defaultColumns);
 
-  const didDefaultColumnChange = useMemo(
-    () => !isEqual(defaultColumns, defaultColumnsRef.current),
-    [defaultColumns]
+  const didDefaultColumnChange = defaultColumns !== defaultColumnsRef.current;
+
+  const setColumnsByColumnIds = useCallback(
+    (columnIds: string[]) => {
+      setVisibleColumns(columnIds);
+      persist({
+        id,
+        storage,
+        storageAlertsTable,
+        columns,
+        visibleColumns: columnIds,
+      });
+    },
+    [columns, id, storage, storageAlertsTable]
   );
 
   useEffect(() => {
     // if defaultColumns have changed,
     // get the latest columns provided by client and
-    if (didDefaultColumnChange) {
+    if (didDefaultColumnChange && defaultColumnsRef.current) {
       defaultColumnsRef.current = defaultColumns;
       setColumnsPopulated(false);
+      // storageAlertsTable = storageData ?? defaultColumns
```

Comment:
Is this still needed?

Reply:
Actually this is supposed to be comment about the structure of `storageAlertTable` because it is not very obvious in the code. But I will change comment to look like a comment and not a code.